### PR TITLE
Clone dataset

### PR DESCRIPTION
**Pull request by zaucker**
Also refactoring create-dataset to avoid code duplication.

### CODE REVIEW

File: lib/Agrammon/DB/Dataset.pm6 [hunk 1]
```diff
@@ -85,31 +96,62 @@ class Agrammon::DB::Dataset does Agrammon::DB {
...
-    has Int  $.records;
+    has Str  $.version is default('2.0-stage'); # TODO: get from config
+    has Int  $.records; # TODO: is this needed?
     has DateTime $.mod-date;
     has $.data;
     has Agrammon::DB::Tag  @.tags;
     has Agrammon::DB::User $.user;
 
-    method create {
+    method create-dataset( $dataset-name, $username, $version, $model ) {
```
Comment (jnthn) [inline lib/Agrammon/DB/Dataset.pm6:106]: If this is factored out for the use of `create` and `clone`, maybe it should be a private method (`method !create-dataset(...)` and the call it as `self!create-dataset(...)`).

File: lib/Agrammon/DB/Dataset.pm6 [hunk 2]
```diff
@@ -133,17 +175,11 @@ class Agrammon::DB::Dataset does Agrammon::DB {
...
-    method !clone($new) {
-        my $dataset = Agrammon::DB::Dataset.new(:name($new));
-        warn "clone($new) not yet implemented";
-        return $dataset;
     }
 
     method submit($email) {
-        my $new-dataset = self!clone("Clone of $!name");
+        my $new-dataset = 'newDataset';
+#        = self!clone("Clone of $!name");
```
Comment (jnthn) [inline lib/Agrammon/DB/Dataset.pm6:182]: Intentionally left for later?

File: lib/Agrammon/Web/Routes.pm6 [hunk 3]
```diff
@@ -105,7 +105,24 @@ sub api-routes (Str $schema, $ws) {
                 }
             }
         }
-        # working
+        operation 'cloneDataset', -> LoggedIn $user {
+            request-body -> ( :oldUsername($old-username), :newUsername($new-username),
```
Comment (jnthn) [inline lib/Agrammon/Web/Routes.pm6:109]: Should one really be able to specify the username of the user that this is cloned to from the outside? Is creating a dataset on any another user's account a desired feature, or could it be a security issue?

Comment (zaucker) [reply lib/Agrammon/Web/Routes.pm6:109]: This is used for giving a copy of one‘s own dataset(s) to another user.

Comment (zaucker) [reply lib/Agrammon/Web/Routes.pm6:109]: In the other hand, I‘ll have to check that one cannily give one‘s own datasets away.

File: lib/Agrammon/DB/Dataset.pm6 (hunk 1 above)
Comment (zaucker) [reply lib/Agrammon/DB/Dataset.pm6:106]: Done.

File: lib/Agrammon/DB/Dataset.pm6 (hunk 2 above)
Comment (zaucker) [reply lib/Agrammon/DB/Dataset.pm6:182]: Yes, made it a bit clearer.

**Review by jnthn — APPROVED** [review]

**Pull request merged**